src/data/int/gcd.lean

```diff
@@ -320,6 +334,36 @@ Compare with `is_coprime.dvd_of_dvd_mul_right` and
 lemma dvd_of_dvd_mul_right_of_gcd_one {a b c : ℤ} (habc : a ∣ b * c) (hab : gcd a b = 1) : a ∣ c :=
 by { rw mul_comm at habc, exact dvd_of_dvd_mul_left_of_gcd_one habc hab }
 
+/-- For nonzero integers `a` and `b`, `gcd a b` is the smallest positive natural number that can be
+written in the form `a * x + b * y` for some pair of integers `x` and `y` -/
+theorem gcd_least_linear {a b : ℤ} (ha : a ≠ 0) :
+  is_least { n : ℕ | 0 < n ∧ ∃ x y : ℤ, ↑n = a * x + b * y } (a.gcd b) :=
+begin
+  simp_rw ←gcd_dvd_iff,
+  split,
+  { simpa [and_true, dvd_refl, set.mem_set_of_eq] using gcd_pos_of_non_zero_left b ha },
+  { simp only [lower_bounds, and_imp, set.mem_set_of_eq],
+    exact λ n hn_pos hn, nat.le_of_dvd hn_pos hn },
+end
+
+lemma gcd_one_iff {a b : ℤ} : gcd a b = 1 ↔ ∃ x y : ℤ, 1 = a * x + b * y :=
+by simpa only [←nat.dvd_one] using @gcd_dvd_iff a b 1
+
+lemma gcd_one_of_dvd_gcd {a b d : ℤ} (ha : a ≠ 0) (h : d = gcd a b) :
+  gcd (a/d) (b/d) = 1 :=
+begin
+  have h_dnz : d ≠ 0, { rw h, exact coe_nat_ne_zero_iff_pos.mpr (gcd_pos_of_non_zero_left b ha) },
+  have hda : d ∣ a, { rw h, exact gcd_dvd_left a b },
+  have hdb : d ∣ b, { rw h, exact gcd_dvd_right a b },
+  rw gcd_one_iff,
+  use [a.gcd_a b, a.gcd_b b],
+  suffices :  d * 1 = d * (a / d * a.gcd_a b + b / d * a.gcd_b b),
+  { exact (mul_right_inj' h_dnz).mp this },
+  simp only [mul_one, mul_add, ←mul_assoc, ←mul_assoc],
+  rw [int.mul_div_cancel' hda, int.mul_div_cancel' hdb, h],
+  apply gcd_eq_gcd_ab,
+end
```

Comment:
I'm not sure this is useful, since the proof could be:
```suggestion
by rw [h, gcd_div_gcd_div_gcd (gcd_pos_of_non_zero_left b ha)]
```

Reply:
In general, any lemma that takes a `x = f y` (like `d = a.gcd b`) hypothesis where x is a variable should probably instead be stated with all the `x`s replaced with `f y`